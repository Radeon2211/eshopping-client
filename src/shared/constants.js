import * as Yup from 'yup';
import { getCountries } from 'country-fns';

export const anyStringRule = Yup.string().min(1).required();

export const userRules = {
  email: Yup.string().email().max(320).trim().required(),
  hideEmail: Yup.bool(),
  username: Yup.string().min(3).max(20).trim().required(),
  password: Yup.string().min(7).max(64).trim().required(),
  firstName: Yup.string().max(60).trim().required(),
  lastName: Yup.string().max(80).trim().required(),
  phonePrefix: Yup.object()
    .shape({
      value: Yup.string().required(),
      label: Yup.string().required(),
    })
    .nullable()
    .required(),
  phoneNumber: Yup.string()
    .min(5)
    .max(15)
    .matches(/^[\d-]{5,15}$/)
    .trim()
    .required(),
  hidePhone: Yup.bool(),
  street: Yup.string().max(60).trim().required(),
  zipCode: Yup.string().max(12).trim().required(),
  city: Yup.string().max(100).trim().required(),
  country: Yup.object()
    .shape({
      value: Yup.string().required(),
      label: Yup.string().required(),
    })
    .nullable()
    .required(),
};

export const productConditions = {
  NEW: 'new',
  USED: 'used',
  NOT_APPLICABLE: 'not_applicable',
};

export const productRules = {
  name: Yup.string().max(150).trim().required(),
  price: Yup.number().min(0.01).max(1000000).required(),
  quantity: Yup.number().min(1).max(100000).required(),
  condition: Yup.string()
    .oneOf([productConditions.NEW, productConditions.USED, productConditions.NOT_APPLICABLE])
    .required(),
  description: Yup.string().max(600).trim(),
};

export const listOfAreaCodes = getCountries().map(({ name, dial }) => {
  const finalValue = `+${dial} ${name.split('(')[0].trim()}`;
  return {
    value: dial,
    label: finalValue,
  };
});

export const listOfCountries = getCountries().map(({ name }) => {
  const finalValue = name.split('(')[0].trim();
  return {
    value: finalValue,
    label: finalValue,
  };
});

export const productPages = {
  ALL_PRODUCTS: 'ALL_PRODUCTS',
  MY_PRODUCTS: 'MY_PRODUCTS',
  USER_PRODUCTS: 'USER_PRODUCTS',
};

export const itemTypes = {
  CART: 'CART',
  ORDER: 'ORDER',
  TRANSACTION: 'TRANSACTION',
};

export const listItemTypes = {
  PRODUCT: 'PRODUCT',
  ORDER: 'ORDER',
};

export const orderTypes = {
  PLACED_ORDERS: 'PLACED_ORDERS',
  SELL_HISTORY: 'SELL_HISTORY',
};

export const inputKinds = {
  INPUT: 'INPUT',
  TEXTAREA: 'TEXTAREA',
  SELECT: 'SELECT',
  RADIO: 'RADIO',
};

export const defaultProductsPerPage = 10;

export const defaultOrdersPerPage = 6;

export const defaultAppPath = '/products?p=1';

export const filtersActions = {
  INIT_STATE: 'INIT_STATE',
  SET_SORT_BY: 'SET_SORT_BY',
  SET_MIN_PRICE: 'SET_MIN_PRICE',
  SET_MAX_PRICE: 'SET_MAX_PRICE',
  SET_CONDITION: 'SET_CONDITION',
};

export const sliderPositionsActions = {
  SET_BOTH: 'SET_BOTH',
  SET_LEFT: 'SET_LEFT',
  SET_RIGHT: 'SET_RIGHT',
};

export const modalTypes = {
  ABOUT_WEBSITE: 'ABOUT_WEBSITE',
  ADD_ADMIN: 'ADD_ADMIN',
  ADD_PRODUCT: 'ADD_PRODUCT',
  BUY_PRODUCTS: 'BUY_PRODUCTS',
  CART_ITEM_ADDED: 'CART_ITEM_ADDED',
  CHANGE_NAME: 'CHANGE_NAME',
  CHANGE_EMAIL: 'CHANGE_EMAIL',
  CHANGE_USERNAME: 'CHANGE_USERNAME',
  CHANGE_PHONE_NUMBER: 'CHANGE_PHONE_NUMBER',
  CHANGE_ADDRESS: 'CHANGE_ADDRESS',
  CHANGE_PASSWORD: 'CHANGE_PASSWORD',
  CHANGE_CONTACTS: 'CHANGE_CONTACTS',
  CHANGE_DELIVERY_ADDRESS: 'CHANGE_DELIVERY_ADDRESS',
  CLEAR_CART: 'CLEAR_CART',
  DELETE_ACCOUNT: 'DELETE_ACCOUNT',
  DELETE_PRODUCT: 'DELETE_PRODUCT',
  EDIT_PRODUCT: 'EDIT_PRODUCT',
  LOGIN: 'LOGIN',
  REMOVE_ADMIN: 'REMOVE_ADMIN',
  RESET_PASSWORD: 'RESET_PASSWORD',
  PENDING_USER_INFO: 'PENDING_USER_INFO',
  SEND_VERIFICATION_LINK: 'SEND_VERIFICATION_LINK',
  SIGNUP: 'SIGNUP',
};

export const sortProductsOptions = [
  { value: undefined, label: 'Default sorting' },
  { value: 'price:asc', label: `Price - ascending` },
  { value: 'price:desc', label: 'Price - descending' },
  { value: 'name:asc', label: 'Name - A to Z' },
  { value: 'name:desc', label: 'Name - Z to A' },
];

export const sortOrdersOptions = [
  { value: 'createdAt:desc', label: 'Date - from newest' },
  { value: 'createdAt:asc', label: 'Date - from oldest' },
  { value: 'overallPrice:asc', label: `Price - ascending` },
  { value: 'overallPrice:desc', label: 'Price - descending' },
];

export const singleInfoNames = {
  USERNAME: 'Username',
  NAME: 'Name',
  EMAIL: 'Email',
  PHONE_NUMBER: 'Phone number',
  ADDRESS: 'Address',
  CONTACTS: 'Contacts',
};

export const updateCartActions = {
  INCREMENT: 'INCREMENT',
  DECREMENT: 'DECREMENT',
  NUMBER: 'NUMBER',
};

export const defaultErrorMessage = 'Something went wrong';

export const productsPerPageControllerOptions = [
  { value: 1, label: '1' },
  { value: 2, label: '2' },
  { value: 3, label: '3' },
  { value: 4, label: '4' },
  { value: 5, label: '5' },
  { value: 10, label: '10' },
  { value: 15, label: '15' },
  { value: 20, label: '20' },
  { value: 25, label: '25' },
];

export const defaultScrollToConfig = {
  top: 0,
  left: 0,
  behavior: 'smooth',
};

export const userStatuses = {
  ACTIVE: 'active',
  PENDING: 'pending',
};
