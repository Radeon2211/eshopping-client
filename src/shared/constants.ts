import * as Yup from 'yup';
import countryCodes from 'country-codes-list';
import { ProductCondition, ProductPhotoFieldValue } from './types/enums';

export const anyStringRule = Yup.string().min(1).required();

export const userRules = {
  email: Yup.string().email().max(320).trim().required(),
  hideEmail: Yup.bool(),
  username: Yup.string().min(3).max(20).trim().required(),
  password: Yup.string().min(7).max(64).trim().required(),
  firstName: Yup.string().max(60).trim().required(),
  lastName: Yup.string().max(80).trim().required(),
  phonePrefix: Yup.object()
    .shape({
      value: Yup.string().required(),
      label: Yup.string().required(),
    })
    .nullable()
    .required(),
  phoneNumber: Yup.string()
    .min(5)
    .max(15)
    .matches(/^[\d-]{5,15}$/)
    .trim()
    .required(),
  hidePhone: Yup.bool(),
  street: Yup.string().max(60).trim().required(),
  zipCode: Yup.string().max(12).trim().required(),
  city: Yup.string().max(100).trim().required(),
  country: Yup.object()
    .shape({
      value: Yup.string().required(),
      label: Yup.string().required(),
    })
    .nullable()
    .required(),
};

export const productRules = {
  name: Yup.string().max(150).trim().required(),
  price: Yup.number().min(0.01).max(1000000).required(),
  quantity: Yup.number().min(1).max(100000).required(),
  condition: Yup.string()
    .oneOf([ProductCondition.NEW, ProductCondition.USED, ProductCondition.NOT_APPLICABLE])
    .required(),
  description: Yup.string().max(600).trim(),
  photo: Yup.mixed()
    .test('is-error', (value) => value !== ProductPhotoFieldValue.ERROR)
    .nullable(),
};

export const listOfAreaCodes = countryCodes.customArray({
  value: '{countryCallingCode}',
  label: '+{countryCallingCode} {countryNameEn}',
});

export const listOfCountries = countryCodes.customArray({
  value: '{countryNameEn}',
  label: '{countryNameEn}',
});

export const itemTypes = {
  CART: 'CART',
  ORDER: 'ORDER',
  TRANSACTION: 'TRANSACTION',
};

export const listItemTypes = {
  PRODUCT: 'PRODUCT',
  ORDER: 'ORDER',
};

export const inputKinds = {
  INPUT: 'INPUT',
  TEXTAREA: 'TEXTAREA',
  SELECT: 'SELECT',
  RADIO: 'RADIO',
};

export const defaultProductsPerPage = 10;

export const defaultOrdersPerPage = 6;

export const defaultAppPath = '/products?p=1';

export const filtersActions = {
  INIT_STATE: 'INIT_STATE',
  SET_SORT_BY: 'SET_SORT_BY',
  SET_MIN_PRICE: 'SET_MIN_PRICE',
  SET_MAX_PRICE: 'SET_MAX_PRICE',
  SET_CONDITION: 'SET_CONDITION',
};

export const sliderPositionsActions = {
  SET_BOTH: 'SET_BOTH',
  SET_LEFT: 'SET_LEFT',
  SET_RIGHT: 'SET_RIGHT',
};

export const sortProductsOptions = [
  { value: undefined, label: 'Default sorting' },
  { value: 'price:asc', label: 'Price - ascending' },
  { value: 'price:desc', label: 'Price - descending' },
  { value: 'name:asc', label: 'Name - A to Z' },
  { value: 'name:desc', label: 'Name - Z to A' },
];

export const sortOrdersOptions = [
  { value: 'createdAt:desc', label: 'Date - from newest' },
  { value: 'createdAt:asc', label: 'Date - from oldest' },
  { value: 'overallPrice:asc', label: 'Price - ascending' },
  { value: 'overallPrice:desc', label: 'Price - descending' },
];

export const singleInfoNames = {
  USERNAME: 'Username',
  NAME: 'Name',
  EMAIL: 'Email',
  PHONE_NUMBER: 'Phone number',
  ADDRESS: 'Address',
  CONTACTS: 'Contacts',
};

export const defaultErrorMessage = 'Something went wrong';

export const productsPerPageControllerOptions = [
  { value: 1, label: '1' },
  { value: 2, label: '2' },
  { value: 3, label: '3' },
  { value: 4, label: '4' },
  { value: 5, label: '5' },
  { value: 10, label: '10' },
  { value: 15, label: '15' },
  { value: 20, label: '20' },
  { value: 25, label: '25' },
];

export const defaultScrollToConfig: ScrollToOptions = {
  top: 0,
  left: 0,
  behavior: 'smooth',
};
