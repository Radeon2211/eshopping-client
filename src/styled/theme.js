const theme = {
  colors: {
    light1: '#FAFAFA',
    light1Transparent: '#FAFAFA88',
    light2: '#DADFE1',
    light3: '#BFBFBF',
    light4: '#666',
    blue: '#1E8BC3',
    blueLight: '#22A7F0',
    blueDark: '#67809F',
    green: '#00A790',
    greenDark: '#009680',
    red: '#F22613',
  },
  shadows: {
    level1: '0 0 2px rgba(0, 0, 0, .1)',
    level2: '0 0 2px rgba(0, 0, 0, .3)',
    level3: '0 0 3px rgba(0, 0, 0, .3)',
    level4: '0 0 5px rgba(0, 0, 0, .3)',
    level5: '0 0 8px rgba(0, 0, 0, .3)',
  },
  fonts: {
    logo: 'Lobster, cursive',
    heading: 'Ubuntu, sans-serif',
    text: 'Lato, sans-serif',
  },
  spacings: {
    level1: '0.8rem',
    level2: '1.2rem',
    level3: '2.4rem',
    level4: '3.6rem',
    level5: '4.8rem',
  },
  zIndexes: {
    level1: 10,
    level2: 100,
    level3: 1000,
    level4: 10000,
  },
};

export default theme;
