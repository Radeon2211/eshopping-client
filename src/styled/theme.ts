const theme = {
  colors: {
    light1: '#FAFAFA',
    light1Transparent: '#FAFAFA88',
    light2: '#DADFE1',
    light3: '#BFBFBF',
    light4: '#777',
    blue: '#1E8BC3',
    blueLight: '#22A7F0',
    blueDark: '#67809F',
    green: '#06AD96',
    greenDark: '#009680',
    red: '#F22613',
    redLight: '#F94D2A',
    redDark: '#985A5A',
  },
  shadows: {
    level1: '0 0 3px rgba(0, 0, 0, 0.03)',
    level2: '0 0 3px rgba(0, 0, 0, 0.3)',
    level3: '0 0 5px rgba(0, 0, 0, 0.3)',
  },
  fonts: {
    logo: 'Lobster, cursive',
    heading: 'Noto Sans, sans-serif',
    text: 'Nunito, sans-serif',
  },
  spacings: {
    level1: '0.8rem',
    level2: '1.2rem',
    level3: '2.4rem',
    level4: '3.6rem',
    level5: '4.8rem',
  },
  zIndexes: {
    level1: 10,
    level2: 100,
    level3: 1000,
    level4: 10000,
    level5: 100000,
  },
  fontSizes: {
    level1: '1.3rem',
    level2: '1.4rem',
    level3: '1.5rem',
    level4: '1.7rem',
    level5: '2.2rem',
    level6: '2.6rem',
  },
  durations: {
    level1: 0.1,
    level2: 0.2,
    level3: 0.3,
  },
  lineHeights: {
    level1: '1',
    level2: '1.1',
    level3: '1.2',
    level4: '1.3',
    level5: '1.4',
    level6: '1.5',
  },
};

export type ThemeSpacing = keyof typeof theme.spacings;
export type ThemeZIndex = keyof typeof theme.zIndexes;
export type ThemeFontSize = keyof typeof theme.fontSizes;
export type ThemeDuration = keyof typeof theme.durations;
export type ThemeLineHeight = keyof typeof theme.lineHeights;

export default theme;
