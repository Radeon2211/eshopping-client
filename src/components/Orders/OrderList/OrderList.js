import PropTypes from 'prop-types';
import { useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import * as SC from './OrderList.sc';
import theme from '../../../styled/theme';
import { GreenText } from '../../../styled/components';
import TransactionAndOrderProdItem from '../../TransactionAndOrderProdItem/TransactionAndOrderProdItem';
import PlainText from '../../UI/PlainText';
import FlexWrapper from '../../UI/FlexWrapper';
import Button from '../../UI/Button/Button';
import LoadingOverlay from '../../UI/LoadingOverlay/LoadingOverlay';
import { formatPrice, formatOrderDate } from '../../../shared/utility/utility';
import * as propTypes from '../../../shared/propTypes';
import { OrderType } from '../../../shared/types/types';

export default function OrderList({ orders, orderType }) {
  const isDataLoading = useSelector((state) => state.ui.isDataLoading);

  const orderList = orders.map((order) => {
    const username =
      orderType === OrderType.PLACED_ORDERS ? order.seller?.username : order.buyer?.username;
    const userType = orderType === OrderType.PLACED_ORDERS ? 'seller ' : 'buyer ';

    return (
      <SC.SingleOrder key={order._id} data-testid="OrderList-single-order">
        <FlexWrapper $justify="space-between" $align="flex-end" $spacing="3">
          <PlainText $size="3">
            <span data-testid="OrderList-user-type">{userType}</span>
            {username ? (
              <Link to={`/user/${username}?p=1`} data-testid="OrderList-user-link">
                <GreenText>{username}</GreenText>
              </Link>
            ) : (
              <PlainText $fStyle="italic" data-testid="OrderList-account-deleted">
                (account has been deleted)
              </PlainText>
            )}
          </PlainText>
          <PlainText $size="2" data-testid="OrderList-created-at-date">
            {formatOrderDate(order.createdAt)}
          </PlainText>
        </FlexWrapper>
        {order.products.map((product) => (
          <TransactionAndOrderProdItem key={product._id} data={product} orderId={order._id} />
        ))}
        <FlexWrapper $mgTop="3" $justify="space-between" $align="center" $spacing="3">
          <Link to={`/order/${order._id}`} data-testid="OrderList-order-details-link">
            <Button>details</Button>
          </Link>
          <div>
            <PlainText $size="3" $mgRight="1" $spacing="1px" $color={theme.colors.light4}>
              TOTAL
            </PlainText>
            <PlainText $size="5" data-testid="OrderList-overall-price">
              {formatPrice(order.overallPrice)}
            </PlainText>
          </div>
        </FlexWrapper>
      </SC.SingleOrder>
    );
  });

  const loadingOverlay = isDataLoading ? <LoadingOverlay /> : null;

  return (
    <SC.Wrapper>
      {orderList}
      {loadingOverlay}
    </SC.Wrapper>
  );
}

OrderList.propTypes = {
  orders: PropTypes.arrayOf(PropTypes.shape(propTypes.orderItem)).isRequired,
  orderType: PropTypes.oneOf(Object.values(OrderType)).isRequired,
};
