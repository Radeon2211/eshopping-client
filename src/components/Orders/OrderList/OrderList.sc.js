import styled from 'styled-components';

export const List = styled.div`
  position: relative;
`;

export const SingleOrder = styled.div`
  padding: ${({ theme }) => theme.spacings.level3} 0;
  border-bottom: 1px solid ${({ theme }) => theme.colors.light3};

  &:first-child {
    padding-top: 0;
  }

  & .username {
    font-size: ${({ theme }) => theme.fontSizes.level3};
  }

  & .date {
    font-size: ${({ theme }) => theme.fontSizes.level2};
  }

  & .total-text {
    font-size: ${({ theme }) => theme.fontSizes.level3};
    letter-spacing: 1px;
    margin-right: ${({ theme }) => theme.spacings.level1};
  }

  & .price {
    font-size: ${({ theme }) => theme.fontSizes.level5};
  }
`;
