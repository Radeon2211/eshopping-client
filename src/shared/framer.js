import theme from '../styled/theme';

export const stepFormVariants = {
  hidden: {
    opacity: 0,
  },
  visible: {
    opacity: 1,
    transition: { duration: theme.durations.level1, delay: theme.durations.level1 },
  },
};

export const backdropVariants = {
  hidden: {
    opacity: 0,
    pointerEvents: 'none',
  },
  visible: {
    opacity: 1,
    pointerEvents: 'initial',
    transition: { duration: theme.durations.level3 },
  },
};

export const modalVariants = {
  hidden: {
    opacity: 0,
    pointerEvents: 'none',
  },
  visible: {
    opacity: 1,
    transition: { delay: theme.durations.level2 },
    pointerEvents: 'initial',
  },
};

export const messageBoxVariants = {
  hidden: {
    pointerEvents: 'none',
    y: '-150%',
  },
  visible: {
    transition: { type: 'spring', duration: theme.durations.level2, stiffness: 100 },
    pointerEvents: 'initial',
    y: 0,
  },
};

export const dropdownVariants = {
  hidden: {
    opacity: 0,
    pointerEvents: 'none',
    y: '-10%',
  },
  visible: {
    opacity: 1,
    pointerEvents: 'initial',
    y: '0',
  },
};
