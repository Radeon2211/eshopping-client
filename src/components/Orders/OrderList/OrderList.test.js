import React from 'react';
import { render, cleanup, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { Router } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import OrderList from './OrderList';
import theme from '../../../styled/theme';
import {
  checkProps,
  createTransactionAndOrderProdItem,
  createOrder,
} from '../../../shared/testUtility';
import { orderTypes } from '../../../shared/constants';

const mockStore = configureMockStore([thunk]);

const setUp = (orders, orderType, isDataLoading = false, pushFn) => {
  const history = {
    listen: jest.fn(),
    createHref: jest.fn(),
    location: { pathname: '/my-account/placed-orders', search: '?p=1' },
    push: pushFn,
  };

  const store = mockStore({
    ui: { isDataLoading },
  });

  return render(
    <Router history={history}>
      <Provider store={store}>
        <ThemeProvider theme={theme}>
          <OrderList orders={orders} orderType={orderType} />
        </ThemeProvider>
      </Provider>
    </Router>,
  );
};

afterEach(cleanup);

describe('<OrderList />', () => {
  describe('Check prop types', () => {
    it('Should NOT throw a warning', () => {
      const props = {
        orders: [createOrder()],
        orderType: orderTypes.PLACED_ORDERS,
      };
      expect(checkProps(OrderList, props)).toBeUndefined();
    });

    it('Should throw a warning', () => {
      expect(checkProps(OrderList, {})).not.toBe(null);
    });
  });

  describe('Check how renders', () => {
    it('Should render everything correctly with two items and type PLACED_ORDERS', () => {
      const products = [
        createTransactionAndOrderProdItem('p1', 'sellerUser', 4, 4, 'product1', true),
        createTransactionAndOrderProdItem('p2', 'sellerUser', 2, 6.4, 'product2'),
      ];
      const orders = [
        createOrder(products, 'o1', 'sellerUser', 'buyerUser', 28.8, '2021-01-08T11:08:51.008Z'),
      ];
      const { asFragment } = setUp(orders, orderTypes.PLACED_ORDERS);
      expect(asFragment()).toMatchSnapshot();
    });

    it('Should render everything correctly with one item and type SELL_HISTORY', () => {
      const products = [
        createTransactionAndOrderProdItem('p1', 'sellerUser', 2, 6.4, 'product1', true),
      ];
      const orders = [
        createOrder(products, 'o1', 'sellerUser', 'buyerUser', 12.8, '2021-02-17T01:53:45.008Z'),
      ];
      const { asFragment } = setUp(orders, orderTypes.SELL_HISTORY);
      expect(asFragment()).toMatchSnapshot();
    });

    it('Should render account deleted - type PLACED_ORDERS', () => {
      const products = [createTransactionAndOrderProdItem('p1', null, 2, 6.4, 'product1', true)];
      const orders = [
        createOrder(products, 'o1', null, 'buyerUser', 12.8, '2021-02-28T21:13:05.008Z'),
      ];
      setUp(orders, orderTypes.PLACED_ORDERS);

      expect(screen.getByTestId('OrderList-account-deleted')).toBeInTheDocument();
      expect(screen.queryByTestId('OrderList-user-link')).not.toBeInTheDocument();
    });

    it('Should render account deleted - type SELL_HISTORY', () => {
      const products = [
        createTransactionAndOrderProdItem('p1', 'sellerUser', 2, 6.4, 'product1', true),
      ];
      const orders = [
        createOrder(products, 'o1', 'sellerUser', null, 12.8, '2021-02-28T21:13:05.008Z'),
      ];
      setUp(orders, orderTypes.SELL_HISTORY);

      expect(screen.getByTestId('OrderList-account-deleted')).toBeInTheDocument();
      expect(screen.queryByTestId('OrderList-user-link')).not.toBeInTheDocument();
    });

    it('Should render <LoadingOverlay />', () => {
      const products = [
        createTransactionAndOrderProdItem('p1', 'sellerUser', 2, 6.4, 'product1', true),
      ];
      const orders = [
        createOrder(products, 'o1', 'sellerUser', 'buyerUser', 12.8, '2021-02-28T21:13:05.008Z'),
      ];
      setUp(orders, orderTypes.PLACED_ORDERS, true);

      expect(screen.getByTestId('LoadingOverlay')).toBeInTheDocument();
    });
  });

  describe('Check behaviour after clicking links', () => {
    it('Should call push with correct path after clicking links - type PLACED_ORDERS', () => {
      const products = [createTransactionAndOrderProdItem()];
      const orders = [createOrder(products, 'o1', 'sellerUser', 'buyerUser')];
      const pushFn = jest.fn();
      setUp(orders, orderTypes.PLACED_ORDERS, false, pushFn);

      fireEvent.click(screen.getByTestId('OrderList-user-link'));
      expect(pushFn).toHaveBeenCalledWith('/user/sellerUser?p=1');

      fireEvent.click(screen.getByTestId('OrderList-order-details-link'));
      expect(pushFn).toHaveBeenLastCalledWith('/order/o1');

      expect(pushFn).toHaveBeenCalledTimes(2);
    });

    it('Should call push with correct path after clicking links - type SELL_HISTORY', () => {
      const products = [createTransactionAndOrderProdItem()];
      const orders = [createOrder(products, 'o1', 'sellerUser', 'buyerUser')];
      const pushFn = jest.fn();
      setUp(orders, orderTypes.SELL_HISTORY, false, pushFn);

      fireEvent.click(screen.getByTestId('OrderList-user-link'));
      expect(pushFn).toHaveBeenCalledWith('/user/buyerUser?p=1');

      fireEvent.click(screen.getByTestId('OrderList-order-details-link'));
      expect(pushFn).toHaveBeenLastCalledWith('/order/o1');

      expect(pushFn).toHaveBeenCalledTimes(2);
    });
  });
});
