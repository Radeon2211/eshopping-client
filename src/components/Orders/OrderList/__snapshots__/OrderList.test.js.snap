// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderList /> check how renders should render everything correctly with one item and type SELL_HISTORY 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ VfWiw"
  >
    <div
      class="sc-gsTCUz bKCzp"
    >
      <div
        class="sc-eCssSg gaIytV"
        direction="row"
        spacing="3"
        width="100%"
        wrap="nowrap"
      >
        <span
          class="sc-pFZIQ bMGSTl"
          color=""
          display="inline-block"
          overflow="visible"
          size="3"
          spacing="0px"
          transform=""
        >
          <span>
            buyer 
          </span>
          <a
            data-testid="OrderList-user-link"
          >
            <span
              class="sc-dlfnbm cJaBxc"
            >
              buyerUser
            </span>
          </a>
        </span>
        <span
          class="sc-pFZIQ cMioVI"
          color=""
          display="inline-block"
          overflow="visible"
          size="2"
          spacing="0px"
          transform=""
        >
          17 Feb 2021, 02:53
        </span>
      </div>
      <div
        class="sc-eCssSg gnpzZR sc-jSgupP iyBJDZ"
        data-testid="TransactionAndOrderProdItem"
        direction="row"
        spacing="2"
        width="100%"
        wrap="nowrap"
      >
        <a
          data-testid="TransactionAndOrderProdItem-product-link-photo"
        >
          <div
            class="sc-eCssSg cldNiM sc-fubCfw jcXzpB"
            data-testid="ProductThumbnail"
            direction="row"
            height="5"
            spacing=""
            width="5"
            wrap="nowrap"
          >
            <img
              alt="product1"
              data-testid="ProductThumbnail-img"
              src="http://192.168.1.109:4000/orders/o1/p1/photo"
            />
          </div>
        </a>
        <div
          class="sc-gKsewC cREdUd"
        >
          <span
            class="name"
          >
            <a
              data-testid="TransactionAndOrderProdItem-product-link-name"
            >
              product1
            </a>
          </span>
          <div
            class="sc-eCssSg bOnlJe sc-iBPRYJ dOEiXN"
            direction="row"
            spacing=""
            width="100%"
            wrap="nowrap"
          >
            <span
              class="sc-pFZIQ eBEFzI"
              color="#777"
              display="inline-block"
              overflow="visible"
              size="2"
              spacing="0px"
              transform=""
            >
              2 x $6.40
            </span>
            <span
              class="sc-pFZIQ iisBUL"
              color=""
              display="inline-block"
              overflow="visible"
              size="4"
              spacing="0px"
              transform=""
            >
              $12.80
            </span>
          </div>
        </div>
      </div>
      <div
        class="sc-eCssSg fEZMEF"
        direction="row"
        spacing="3"
        width="100%"
        wrap="nowrap"
      >
        <a
          data-testid="OrderList-order-details-link"
        >
          <button
            class="sc-jrAGrp iifaoN"
            color="blue"
            type="button"
          >
            details
          </button>
        </a>
        <div>
          <span
            class="sc-pFZIQ tbjjM"
            color="#777"
            display="inline-block"
            overflow="visible"
            size="3"
            spacing="1px"
            transform=""
          >
            TOTAL
          </span>
          <span
            class="sc-pFZIQ ftfIUZ"
            color=""
            display="inline-block"
            overflow="visible"
            size="5"
            spacing="0px"
            transform=""
          >
            $12.80
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<OrderList /> check how renders should render everything correctly with two items and type PLACED_ORDERS 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ VfWiw"
  >
    <div
      class="sc-gsTCUz bKCzp"
    >
      <div
        class="sc-eCssSg gaIytV"
        direction="row"
        spacing="3"
        width="100%"
        wrap="nowrap"
      >
        <span
          class="sc-pFZIQ bMGSTl"
          color=""
          display="inline-block"
          overflow="visible"
          size="3"
          spacing="0px"
          transform=""
        >
          <span>
            seller 
          </span>
          <a
            data-testid="OrderList-user-link"
          >
            <span
              class="sc-dlfnbm cJaBxc"
            >
              sellerUser
            </span>
          </a>
        </span>
        <span
          class="sc-pFZIQ cMioVI"
          color=""
          display="inline-block"
          overflow="visible"
          size="2"
          spacing="0px"
          transform=""
        >
          8 Jan 2021, 12:08
        </span>
      </div>
      <div
        class="sc-eCssSg gnpzZR sc-jSgupP iyBJDZ"
        data-testid="TransactionAndOrderProdItem"
        direction="row"
        spacing="2"
        width="100%"
        wrap="nowrap"
      >
        <a
          data-testid="TransactionAndOrderProdItem-product-link-photo"
        >
          <div
            class="sc-eCssSg cldNiM sc-fubCfw jcXzpB"
            data-testid="ProductThumbnail"
            direction="row"
            height="5"
            spacing=""
            width="5"
            wrap="nowrap"
          >
            <img
              alt="product1"
              data-testid="ProductThumbnail-img"
              src="http://192.168.1.109:4000/orders/o1/p1/photo"
            />
          </div>
        </a>
        <div
          class="sc-gKsewC cREdUd"
        >
          <span
            class="name"
          >
            <a
              data-testid="TransactionAndOrderProdItem-product-link-name"
            >
              product1
            </a>
          </span>
          <div
            class="sc-eCssSg bOnlJe sc-iBPRYJ dOEiXN"
            direction="row"
            spacing=""
            width="100%"
            wrap="nowrap"
          >
            <span
              class="sc-pFZIQ eBEFzI"
              color="#777"
              display="inline-block"
              overflow="visible"
              size="2"
              spacing="0px"
              transform=""
            >
              4 x $4
            </span>
            <span
              class="sc-pFZIQ iisBUL"
              color=""
              display="inline-block"
              overflow="visible"
              size="4"
              spacing="0px"
              transform=""
            >
              $16
            </span>
          </div>
        </div>
      </div>
      <div
        class="sc-eCssSg gnpzZR sc-jSgupP iyBJDZ"
        data-testid="TransactionAndOrderProdItem"
        direction="row"
        spacing="2"
        width="100%"
        wrap="nowrap"
      >
        <a
          data-testid="TransactionAndOrderProdItem-product-link-photo"
        >
          <div
            class="sc-eCssSg cldNiM sc-fubCfw jcXzpB"
            data-testid="ProductThumbnail"
            direction="row"
            height="5"
            spacing=""
            width="5"
            wrap="nowrap"
          >
            <img
              alt="product2"
              data-testid="ProductThumbnail-img"
              src="no-photo.png"
            />
          </div>
        </a>
        <div
          class="sc-gKsewC cREdUd"
        >
          <span
            class="name"
          >
            <a
              data-testid="TransactionAndOrderProdItem-product-link-name"
            >
              product2
            </a>
          </span>
          <div
            class="sc-eCssSg bOnlJe sc-iBPRYJ dOEiXN"
            direction="row"
            spacing=""
            width="100%"
            wrap="nowrap"
          >
            <span
              class="sc-pFZIQ eBEFzI"
              color="#777"
              display="inline-block"
              overflow="visible"
              size="2"
              spacing="0px"
              transform=""
            >
              2 x $6.40
            </span>
            <span
              class="sc-pFZIQ iisBUL"
              color=""
              display="inline-block"
              overflow="visible"
              size="4"
              spacing="0px"
              transform=""
            >
              $12.80
            </span>
          </div>
        </div>
      </div>
      <div
        class="sc-eCssSg fEZMEF"
        direction="row"
        spacing="3"
        width="100%"
        wrap="nowrap"
      >
        <a
          data-testid="OrderList-order-details-link"
        >
          <button
            class="sc-jrAGrp iifaoN"
            color="blue"
            type="button"
          >
            details
          </button>
        </a>
        <div>
          <span
            class="sc-pFZIQ tbjjM"
            color="#777"
            display="inline-block"
            overflow="visible"
            size="3"
            spacing="1px"
            transform=""
          >
            TOTAL
          </span>
          <span
            class="sc-pFZIQ ftfIUZ"
            color=""
            display="inline-block"
            overflow="visible"
            size="5"
            spacing="0px"
            transform=""
          >
            $28.80
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
